include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,330 @@
+   */
+  std::unordered_map<std::string, int> name_to_idx_map_;
+
+  /**
+   * @brief Names of behaviors in `behaviors_`
+   */
+  std::vector<std::string> names_;
+
+  /**
+   * @brief Index in `behaviors_` of the selected behavior
+   */
+  unsigned int selected_idx = 0;
+
+  /**
+   * @brief Index in `behaviors_` of the incoming behavior while transitioning
+   *        behaviors
+   */
+  int next_idx_ = 0;
+
+  /**
+   * @brief Indicates whether a behavior switch is in progress
+   */
+  bool switching_behaviors_ = false;
+
+  /**
+   * @brief Constructs and initializes a behavior
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   * @return unique pointer to the behavior if initialization was successful,
+   *         nullptr otherwise
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  std::unique_ptr<BehaviorType> ConstructBehavior(ConstructorArgs &&... args) {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = std::make_unique<BehaviorType>(args...);
+    if (b->Init()) {
+      b->set_initialized();
+      return std::move(b);
+    } else { return nullptr; }
+  }
+
+ public:
+  /**
+   * @brief Construct a behavior manager object
+   * @param robot pointer to robot that the behavior is executing on
+   */
+  BehaviorManager(std::shared_ptr<Robot> robot) : robot_(robot) {
+    default_name_ = "OFF";
+    AddBehavior<OffBehavior<Robot>>(robot_, default_name_);
+  }
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~BehaviorManager() = default;
+
+  /**
+   * @brief Set the current behavior
+   * @param next_idx index in `behaviors_`
+   */
+  virtual void SetBehavior(const int &next_idx) {
+    if (next_idx == KILL_ROBOT_IDX) {
+      kodlab::CTRL_C_DETECTED = true; // kill robot
+      LOG_FATAL("Behavior %d received, killing robot.", KILL_ROBOT_IDX);
+    } else if (next_idx == selected_idx) {
+      LOG_INFO("Behavior %d is already active.", next_idx);
+      return;
+    } else if (next_idx < behaviors_.size() && next_idx >= 0) {
+      next_idx_ = next_idx; // update next behavior
+      if (!behaviors_[next_idx_]->is_initialized()) {
+        behaviors_[next_idx_]->Init();
+      }
+      behaviors_[selected_idx]->Stop(*behaviors_[next_idx_]); // stop current behavior
+      switching_behaviors_ = true;  // indicate transition is in progress
+    } else {
+      LOG_WARN("Invalid behavior selected: %d", next_idx);
+    }
+  }
+
+  /**
+   * @brief Set the current behavior
+   * @param name name of behavior in `behaviors_`
+   */
+  virtual void SetBehavior(const std::string &name) {
+    if (name == KILL_BEHAVIOR_NAME) { SetBehavior(KILL_ROBOT_IDX); }
+    else {
+      auto idx = get_behavior_index(name);
+      if (idx) { SetBehavior(idx.value()); }
```

Comment:
This is worse as far as formatting than the previous IMO.

Reply:
Formatted per style guide in 57a65848a1b429d69bd735eddc489b60878e210b